lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/job/SwapUpdated.kt

```diff
@@ -54,11 +64,62 @@ class SwapUpdatedJob(
         try {
             val request = Json.decodeFromString(SwapUpdatedRequest.serializer(), payload)
             this.swapIdHash = request.id
+            startPolling(liquidSDK)
         } catch (e: Exception) {
             logger.log(TAG, "Failed to decode payload: ${e.message}", "WARN")
         }
     }
 
+    private fun startPolling(liquidSDK: BindingLiquidSdk) {
+        pollingJob = scope.launch {
+            while (isActive) {
+                try {
+                    if (swapIdHash == null) {
+                        stopPolling(Exception("Missing swap ID"))
+                        return@launch
+                    }
+
+                    liquidSDK.getPayment(GetPaymentRequest.SwapId(swapIdHash!!))?.let { payment ->
+                        when (payment.status) {
+                            PaymentState.COMPLETE -> {
+                                onEvent(SdkEvent.PaymentSucceeded(payment))
+                                stopPolling()
+                                return@launch
+                            }
+                            PaymentState.WAITING_FEE_ACCEPTANCE -> {
+                                onEvent(SdkEvent.PaymentWaitingFeeAcceptance(payment))
+                                stopPolling()
+                                return@launch
+                            }
+                            PaymentState.PENDING -> {
+                                if (paymentClaimIsBroadcasted(payment.details)) {
+                                    onEvent(SdkEvent.PaymentWaitingConfirmation(payment))
+                                    stopPolling()
+                                    return@launch
+                                }
+                            }
+                            else -> { }
+                        }
+                    }
+                    delay(pollingInterval)
+                } catch (e: Exception) {
+                    stopPolling(e)
+                    return@launch
+                }
+            }
+        }
+    }
+
+    private fun stopPolling(error: Exception? = null) {
```

Comment:
Do we also need to stop polling if onShutdown is called?

Reply:
I don't think so. That only happens when the OS wants to clean up the notification service so it would automatically stop polling.